core/templates/dev/head/pages/exploration_editor/editor_tab/StateHints.js

```diff
@@ -115,10 +117,12 @@ oppia.controller('StateHints', [
             $scope.hintIndex = stateHintsService.displayed.length + 1;
 
             $scope.saveHint = function() {
+              var contentId = GenerateContentIdService.generateUniqueId();
```

Comment:
Does this guarantee a repeat key isn't generated?

Reply:
Yep! (see [here](https://github.com/oppia/oppia/pull/5007/files#diff-d1af681a841a205395ad99f4baeb84baR20))